### PR TITLE
Refactor Jamendo to use the ProviderDataIngester

### LINKED ISSUE #587: Refactor Jamendo to use ProviderDataIngester
## Suggested Improvement

Refactor the Jamendo provider script to use the new ProviderDataIngester base class

## Benefit

More details in #229

## Implementation

- [ ] 🙋 I would be interested in implementing this feature.

### PR DESCRIPTION
**Pull request by stacimc**
## Fixes

Fixes #587 by @stacimc 

## Description

Refactors Jamendo to use the ProviderDataIngester base class.

## Testing Instructions

* Run the Jamendo provider script locally
* Try setting an `ingestion_limit`, maybe try setting some of the dagrun conf options like `initial_query_params`
* `just test`

Locally, I ran Jamendo on main and then on this branch and compared the tsvs.

## Checklist

- [ ] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [ ] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
- [ ] My commit messages follow [best practices][best_practices].
- [ ] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [ ] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: openverse_catalog/dags/providers/provider_api_scripts/jamendo.py [hunk 1]
```diff
@@ -16,328 +16,223 @@
...
+            return None
+
+        if (foreign_landing_url := data.get("shareurl")) is None:
+            return None
+
+        if (audio_url := self._get_audio_url(data)) is None:
+            return None
+
+        license_url = data.get("license_ccurl")
+        logger.info(license_url)
```
Comment (AetherUnbound) [inline openverse_catalog/dags/providers/provider_api_scripts/jamendo.py:173]: Should this and other `logger` calls be at `logger.debug`? It seems like this is at least 5 log lines per record which could be a lot.

File: tests/dags/providers/provider_api_scripts/test_jamendo.py [hunk 2]
```diff
@@ -41,168 +46,30 @@ def test_remove_param_from_url(url, param, expected):
...
+    assert jamendo.get_batch_data(None) is None
 
 
-def test_get_image_pages_returns_correctly_with_no_results():
-    expect_result = None
-    with patch.object(jamendo.delayed_requester, "get_response_json", return_value={}):
-        actual_result = jamendo._get_batch_json()
-    assert actual_result == expect_result
+def test_get_batch_data_returns_correctly_with_no_results():
+    assert jamendo.get_batch_data({}) is None
```
Comment (AetherUnbound) [inline tests/dags/providers/provider_api_scripts/test_jamendo.py:54]: We could group these two tests in a parametrization!

File: tests/dags/providers/provider_api_scripts/test_jamendo.py [hunk 3]
```diff
@@ -233,7 +100,49 @@ def test_extract_audio_data_handles_example_dict():
...
+        audio_data = json.load(f)
+        audio_data.pop("audio", None)
+    assert jamendo.get_record_data(audio_data) is None
+
+
+def test_get_record_data_returns_none_when_no_license():
+    with open(RESOURCES / "audio_data_example.json") as f:
+        audio_data = json.load(f)
+        audio_data.pop("license_ccurl", None)
+    assert jamendo.get_record_data(audio_data) is None
```
Comment (AetherUnbound) [inline tests/dags/providers/provider_api_scripts/test_jamendo.py:124]: These 3 tests could also potentially be parametrized.

**Review by AetherUnbound — CHANGES_REQUESTED** [review]
This looks great and works fine locally! I think the log settings need to be changed, but after that this is good to go 💯

**Review by AetherUnbound — APPROVED** [review]
Fantastic!

**Review by krysal — APPROVED** [review]
Looks great!

**Pull request merged**